### PR TITLE
chore(renovate): autocommit pre-commit changes

### LINKED ISSUE #81: Update `README`/`values.schema.json` for Renovate PRs
Renovate updates dependencies. These require us to update the `README` and `values.schema.json` files, the PR can't be merged otherwise. We should update the Renovate PRs with an additional commit populating these changes.

### PR DESCRIPTION
**Pull request by tumido**
## Description of the change

We want renovate to update our dependencies. However any dependency change means need to regenerate metadata files via `pre-commit`. This workflow should allow that.

## Existing or Associated Issue(s)

https://github.com/janus-idp/helm-backstage/issues/81

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [pre-commit](https://pre-commit.com/) utility can be used to generate the necessary content. Use `pre-commit run -a` to apply changes.
- [x] JSON Schema template updated and re-generated the raw schema via `pre-commit` hook.
- [x] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

### CODE REVIEW

**Review by schwesig — APPROVED** [review]
/lgtm
/approve
OK from my side, didn't find any problems

File: .github/workflows/update-renovate-prs.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4
+        with:
+          go-version: ^1
+
+      - name: Setup helm-docs
+        run: go install github.com/norwoodj/helm-docs/cmd/helm-docs@latest
+
+      - name: Run pre-commit
+        uses: pre-commit/action@646c83fcd040023954eafda54b4db0192ce70507 # renovate: tag=v3.0.0
+        continue-on-error: true
```
Comment (RobotSail) [inline .github/workflows/update-renovate-prs.yml:37]: Shouldn't this workflow fail if we can't run the pre-commit hooks here?

File: .github/workflows/update-renovate-prs.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          ref: ${{ github.head_ref }}
+
+      - uses: actions/setup-python@bd6b4b6205c4dbad673328db7b31b7fab9e241c0 # v4
+        with:
+          python-version: 3.7
+
+      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4
+        with:
+          go-version: ^1
```
Comment (RobotSail) [inline .github/workflows/update-renovate-prs.yml:30]: NP: Should we set a specific Go version here? Or will it just always default to whatever is available for the latest?

**Review by RobotSail — APPROVED** [review]
Just a few questions but otherwise LGTM

File: .github/workflows/update-renovate-prs.yml (hunk 2 above)
Comment (tumido) [reply .github/workflows/update-renovate-prs.yml:30]: Yes, we don't care about golang version.

File: .github/workflows/update-renovate-prs.yml (hunk 1 above)
Comment (tumido) [reply .github/workflows/update-renovate-prs.yml:37]: Pre-commit always errors (exit code == 1) if if founds itself changing files. But in this case we want to use it TO change files. So we expect it to return exit code 1 and we want to continue the workflow afterwards. It's a `do_something() || true` kind of a hack.

File: .github/workflows/update-renovate-prs.yml (hunk 2 above)
Comment (RobotSail) [reply .github/workflows/update-renovate-prs.yml:30]: Great

File: .github/workflows/update-renovate-prs.yml (hunk 1 above)
Comment (RobotSail) [reply .github/workflows/update-renovate-prs.yml:37]: Ah I understand. Thank you for the explanation!

**Pull request merged**